multimedia/M2VDownsizer/Files/Makefile

```diff
@@ -0,0 +1,257 @@
+
+# Define library directories other than the system library directory, typically /usr/lib
+#   Statement Format:  LFLAGS := -Llibrary_path
+#   E.g. LIBDIRS := -L/opt/local/lib
+#
+LIBDIRS := -Lobj  
+
+# Define any MacOS Framework directories other than the system Framework directory
+#   Statement Format: FFLAGS:= -Fframework_path
+#   E.g. FWORKDIRS = -F./ -F/System/Library/Frameworks/
+#
+#FWORKDIRS = -F./ -F/System/Library/Frameworks/
+
+# Define any libraries to link into executable:
+#   Statement Format: LIBS := -llibname
+#   E.g. if I want to link the math library, use -lm as follows:
+#   LIBS := -lm
+#
+LIBS := -lmpeg2
+
+# Define any MacOS Frameworks to link into executable:
+#   Statement Format: FWORKS := -framework framework_name
+#   E.g. FWORKS := -framework CoreServices
+#
+FWORKS := -framework CoreServices
+
+# Define the C source code files and directories
+#
+SRCS-UTILS := $(shell find utils ! -path utils -prune -type f -name '*.c' -exec basename {} \;)
+SRCS-MPEG2ENC := $(shell find mpeg2enc -type f -name '*.c' -exec basename {} \;)
+SRCS-ALTIVEC := $(shell find utils/altivec -type f -name '*.c' -exec basename {} \;)
+SRCS-ALTIVEC-ASSM := $(shell find utils/altivec -type f -name '*.S' -exec basename {} \;)
+SRCS-MAIN := $(shell find ./ ! -path ./ -prune -type f -name '*.c' -exec basename {} \;)
+
+SRCDIR-UTILS := utils
+SRCDIR-ALTIVEC := utils/altivec
+SRCDIR-MPEG2ENC := mpeg2enc
+SRCDIR-MAIN := ./
+
+# Define the C object files 
+#
+# This uses Suffix Replacement within a macro:
+#   $(name:string1=string2)
+#         For each word in 'name' replace 'string1' with 'string2'
+# Below we are replacing the suffix .c of all words in the macro SRCS
+# with the .o suffix
+#
+OBJS-UTILS := $(SRCS-UTILS:.c=.o)
+OBJS-ALTIVEC := $(SRCS-ALTIVEC:.c=.o)
+OBJS-ALTIVEC-ASSM:= $(SRCS-ALTIVEC-ASSM:.S=.o)
+OBJS-MPEG2ENC := $(SRCS-MPEG2ENC:.c=.o)
+OBJS-MAIN := $(SRCS-MAIN:.c=.o)
+
+# Define main object directory
+#
+OBJDIR := obj
+
+# Define object sub-directories
+#
+# Do not change the following directories!!
+# The name of the last directory  MUST match the name of the corresponding 
+# source directory in order for makedepend to work!! 
+# (unless the -p option is used with makedepend)
+# 
+OBJDIR-UTILS := $(OBJDIR)/utils
+OBJDIR-ALTIVEC := $(OBJDIR)/altivec
+OBJDIR-MPEG2ENC := $(OBJDIR)/mpeg2enc
+OBJDIR-MAIN := $(OBJDIR)/main
+
+# Define the final executable file(s)
+#
+PRODUCT := M2VDownsizer
+
+# Define the PHONY targets
+#
+.PHONY: all depend clean utils altivec mpeg2enc main objects macports print-
+
+# define the default target
+# The default target is typically the "all" PHONY target
+#
+default: all
+
+# Define the "all" PHONY target
+#
+all: utils altivec mpeg2enc main $(PRODUCT)
+
+# The following also works for the "all" PHONY target but will not display messages 
+# after each subdirectory has been processed (e.g. compiled).
+#
+# all: $(PRODUCT)
+#
+
+# Define the "macports" PHONY target
+#
+macports: depend all
```

Comment:
Does the "macports" target work? In a parallel build scenario, the prerequisites "depend" and "all" will be built in parallel, unless for example the "all" target were to define that the "depend" target is a prerequisite, which it currently doesn't.

Is the "macports" target needed? The Portfile doesn't specify a target so it uses the "all" target.

Reply:
Yes, the `macports` target works.  But if you use it, one will need to add to the Portfile:

```
depends_build       port:makedepend
build.target            macports
```

I elected not to bother running `make depend` since the generation of all the header dependencies are really not needed to build M2VDownsizer from scratch.

Of course, if you were a developer trying to hack or change the source code for M2VDownsizer, then running `make depend` makes sense.  Since running `make` or 'make all` would only re-compile the minimum number of source code files after one has changed a header file (or other file).

P.S.  Changing the build target from `all` to `macports` will not very much time to the build time in Macports, given the size of the project.

Reply:
Ryan, If you are worried that others might get confused regarding the presence of the `macports` target in the Makefile, we could change the name of this target from `macports` to something like `dev-first-build`.  Only two changes would be needed.  In the .PHONY statement and in the target definition itself (and in the comment above).

Or we could just use the `macports` target in the port.  But that would add a build dependency.

Reply:
> Yes, the `macports` target works.

You are certain it does what you want? I assume you wanted `make depend` to finish running before `make all` starts to run but I don't think that is guaranteed to be the case if parallel building is happening.

> Or we could just use the `macports` target in the port. But that would add a build dependency.

Since it works when using just the `all` target I would leave it that way, and in that case I'm not sure that the `macports` target needs to exist at all.

Reply:
Thanks Ryan.  Whenever I updated the port to use the `macports` build target (instead of the default `all` build target), I noticed that Macports always finished the `make depend` target before starting the `make all` target.  Perhaps, that was caused by the design of `make` which normally will not start the next target until the previous target was finished.  Or maybe it was just a fluke.

**In any event, probably the easiest resolution is to delete the `macports` target and the reference to `macports` from `.PHONY` from the Makefile.**   It may be too confusing for others.

This approach would be consistent with my earlier decision to omit the `make install` target from the Makefile since it was not needed for the Macports port package.

Reply:
This issue has been resolved in my last push.  I removed the `macport` target from the Makefile since it is not needed and my just cause confusion to others reading the Makefile.